buf/registry/module/v1beta1/label.proto

```diff
@@ -71,6 +72,55 @@ message Label {
     (buf.validate.field).required = true,
     (buf.validate.field).string.uuid = true
   ];
+  // Whether or not policy checks are enabled for this Label.
+  //
+  // Policy checks are an enterprise-only feature - contact us to learn more!
+  bool policy_checks_enabled = 10;
+  // The PolicyChecksState for the Commit the Label points to.
+  //
+  // This field will be set if either:
+  //   - Policy checks were enabled on this Label when this Commit was pushed.
+  //   - This Commit transitively depends on a Commit that has a PolicyChecksState.
+  PolicyChecksState policy_checks_state = 11;
+}
+
+// The state of a Commit's policy checks for a particular Label.
+//
+// Policy checks are an enterprise-only feature - contact us to learn more!
+message PolicyChecksState {
+  // The status of the policy checks.
+  PolicyChecksStatus status = 1 [(buf.validate.field).enum.defined_only = true];
+
+  // The time the status was last updated.
+  google.protobuf.Timestamp status_updated_at = 3 [(buf.validate.field).required = true];
+
+  // The ids of Commits that have policy checks blocking these policy checks.
+  //
+  // This field will only be set if the status is blocked.
+  repeated string blocked_on_commit_ids = 2 [(buf.validate.field).string.uuid = true];
```

Comment:
This repeated field is in `Commit`, and could be unbounded - if I have 10000 of these, are all of them going to appear on a `Commit`? I'm generally hesitant to have `repeated` fields on Resources, wondering if this is required.

Reply:
We are actually still debating whether we need this or not, so it might go away entirely. In the event that we keep it...

> This repeated field is in Commit
This isn't meant to be a field on `Commit` it is info that needs to be bound to a union of a Commit and a Label (which IIRC we have a table for on the backend but this is not explicitly modeled in the API)

> could be unbounded - if I have 10000 of these, are all of them going to appear on a Commit?

I think the natural bound for this field would be the number of modules that are in a single workspace (because that is the only circumstance where this blocking state would be relevant). Does that constraint this to a scale you are comfortable with?

This is mostly for display purposes in the UI, so I think designing a separate service/entity for paging would be overkill. If this ends up being a problem, I would rather go in the other direction of just documenting that it will contain a maximum of 10-100 items (or whatever limit you are comfortable with). The UI purpose is to show you want might need to be reviewed to get this commit unblocked so it is sufficient to only show some of them if there are a large number.

Reply:
removed this from the PR for now. Will add back in a future PR if we decide to implement blocked status